### PR TITLE
Sync 427 update readme

### PR DESCRIPTION
**Pull request by jdragojevic**
### Pull Request checklist ###
This is a text-only update to the logins component readme.  The goal is to provide a repeatable format that keeps the living details for a component including but not limited to:
- Feature overview
- Business logic
- api documentation
- links to quality and telmetry dashboards
- getting started information
...

Please review in terms of overall content, what key information are we missing that would help someone new to logins understand how to use, test, measure and monitor it.  This information will be linked to directly from the documentation hub.

### CODE REVIEW

File: components/logins/README.md [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
+
+## Implementation Overview
 Logins implements encrypted storage for login records on top of SQLcipher, with
-support for Sync (using the sync15 crate). It uses a slight modification on top
-of the database schema that firefox-ios used, however many of the queries and
-the way the sync is performed are different, as to allow syncs to complete with
-fewer database operations. See the header comment in `src/schema.rs` for an
-overview of the schema.
+support for Sync (using the [sync15](https://github.com/mozilla/application-services/tree/master/components/sync15) crate). It uses a slight modification on top
+of the database schema that [firefox-ios](https://github.com/mozilla-mobile/firefox-ios/blob/faa6a2839abf4da2c54ff1b3291174b50b31ab2c/Storage/SQL/SQLiteLogins.swift) used.  Notable difference include many of the queries and how sync is performed in order to allow syncs to complete with
```
Comment (thomcc) [inline components/logins/README.md]: nit: differences.

Also, possibly worth noting that while the iOS schema used timestamps in microseconds, our timestamps are in milliseconds.

File: components/logins/README.md [hunk 2]
```diff
@@ -15,11 +18,70 @@ The relevant directories are as follows:
...
+
+## Testing
+
+![status-img](https://img.shields.io/static/v1?label=test%20status&message=acceptable&color=darkgreen)
+
+Our goal is to seek an _acceptable_ level of test coverage, which basically means we feel confident that if we are making changes to the component the testing will alert us to related regressions.  When making changes in an area, make an effort to improve (or minimally not reduce) coverage. Test coverage assessment includes:
+* [rust tests](https://github.com/mozilla/application-services/blob/master/testing/sync-test/src/logins.rs)
+* TODO add link [android tests]()
+* TODO add link [ios tests]()
+* TODO add link [ffi tests]
```
Comment (thomcc) [inline components/logins/README.md]: No specific FFI tests exist (they'd be much too difficult and error-prone to write -- I don't think we'd want these even if we have infinite time to author tests), but the FFI is well-covered by the android tests.

File: components/logins/README.md [hunk 3]
```diff
@@ -15,11 +18,70 @@ The relevant directories are as follows:
...
+## API Documentation
+- TODO [Expand and update API docs](https://github.com/mozilla/application-services/issues/1747)
+
+## Testing
+
+![status-img](https://img.shields.io/static/v1?label=test%20status&message=acceptable&color=darkgreen)
+
+Our goal is to seek an _acceptable_ level of test coverage, which basically means we feel confident that if we are making changes to the component the testing will alert us to related regressions.  When making changes in an area, make an effort to improve (or minimally not reduce) coverage. Test coverage assessment includes:
+* [rust tests](https://github.com/mozilla/application-services/blob/master/testing/sync-test/src/logins.rs)
+* TODO add link [android tests]()
```
Comment (thomcc) [inline components/logins/README.md]: https://github.com/mozilla/application-services/tree/master/components/logins/android/src/test/java/mozilla/appservices/logins

File: components/logins/README.md [hunk 4]
```diff
@@ -15,11 +18,70 @@ The relevant directories are as follows:
...
+* TODO add link [ios tests]()
```
Comment (thomcc) [inline components/logins/README.md]: https://github.com/mozilla/application-services/blob/master/megazords/ios/MozillaAppServicesTests/LoginsTests.swift

File: components/logins/README.md [hunk 5]
```diff
@@ -15,11 +20,69 @@ The relevant directories are as follows:
...
   use JNA to call into to the code in `ffi`.
 - `ios`: This contains the iOS binding to logins, written in Swift. These use
   Swift's native support for calling code written in C to call into the code in
   `ffi`.
+
+## Features
+1. Locally encrypted storage of username and password information
+1. Create, Update and Delete (CRUD) operations for login data
+1. Syncing of logins for devices and apps connected via Firefox Accounts
+1. Import functionality from existing login storage (ex: Fx Desktop or Fennec)
```
Comment (linacambridge) [inline components/logins/README.md]: Nit: Let's use Firefox for iOS as an example. We don't import from Desktop (yet!), and you've covered Fennec with data migration below!

File: components/logins/README.md [hunk 6]
```diff
@@ -15,11 +20,69 @@ The relevant directories are as follows:
...
+At any given time records can exist in 3 places, the local storage, the remote record, and the shared parent.  The shared parent refers to a record that has been synced previously and is referred to in the code as the mirror.
+Login records are encrypted and stored locally. For any record that does not have a shared parent the login component tracks that the record has never been synced.
+
+Reference the [Logins chapter of the synconomicon](https://mozilla.github.io/application-services/synconomicon/ch01.1-logins.html) for detailed information on the record storage format.
+
+### Sign-out behavior
+When the user signs out of their Firefox Account, we reset the storage and clear the shared parent.
+
+### Merging records
+When records are added, the logins component performs a three-way merge between the local record, the remote record and the shared parent (last update on the server).  Details on the merging algorithm are contained in the [generic sync rfc](https://github.com/mozilla/application-services/blob/1e2ba102ee1709f51d200a2dd5e96155581a81b2/docs/design/remerge/rfc.md#three-way-merge-algorithm).
```
Comment (linacambridge) [inline components/logins/README.md]: Oooh, nice, I think @thomcc's doc is an excellent framework for understanding how merging works! We may want to clarify which fields get merged how (for example, that we use `take_max` for `timeLastUsed` and `sum` for `timesUsed`), but that doesn't need to be done here.

File: components/logins/README.md [hunk 7]
```diff
@@ -15,11 +20,69 @@ The relevant directories are as follows:
...
+
+### Record de-duplication
+
+De-duplication compares the records for same the username and same url, but with different passwords. De-duplication compares the records for same the username and same url, but with different passwords.  Deduplication logic is based on age, the username and hostname.
```
Comment (linacambridge) [inline components/logins/README.md]: The first sentence is, um...duplicated? 😁

File: components/logins/README.md [hunk 8]
```diff
@@ -15,11 +20,69 @@ The relevant directories are as follows:
...
+- iOS: start with the [guide to consuming rust components on iOS](https://github.com/mozilla/application-services/blob/master/docs/howtos/consuming-rust-components-on-ios.md)
+
+## API Documentation
+- TODO [Expand and update API docs](https://github.com/mozilla/application-services/issues/1747)
+
+## Testing
+
+![status-img](https://img.shields.io/static/v1?label=test%20status&message=acceptable&color=darkgreen)
+
+Our goal is to seek an _acceptable_ level of test coverage, which basically means we feel confident that if we are making changes to the component the testing will alert us to related regressions.  When making changes in an area, make an effort to improve (or minimally not reduce) coverage. Test coverage assessment includes:
```
Comment (linacambridge) [inline components/logins/README.md]: Nit: Run-on sentence.

**Pull request merged**